src/averageAges.js

```diff
@@ -37,7 +40,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const isMother = withChildren
+    ? people.filter(person => person.sex === 'f'
+      && people.some(child => child.mother === person.name))
+    : people.filter(person => person.sex === 'f');
+  const allAges = isMother.map(person => person.died - person.born);
+  const averageAges = allAges.reduce((sum, age) => sum + age) / allAges.length;
```

Comment:
this logic looks similar to the previous function. Consider creating a separate function for this logic